src/aleph/toolkit/batch.py

```diff
@@ -0,0 +1,18 @@
+from typing import AsyncIterator, List, TypeVar
+
+T = TypeVar("T")
+
+
+async def async_batch(
+    async_iterable: AsyncIterator["T"], n: int
+) -> AsyncIterator[List[T]]:
+    batch = []
```

Comment:
I looked at how to make this work without lists, and using Python 3.10 `anext` and `aiter` would do the job well. These are absent from previous versions of Python that we may still want to support 🤔. Or not, since we deploy everything in containers under our control.

This could approximately become (not handling the last batch correctly ?):
```python
from typing import AsyncIterator, List, TypeVar

T = TypeVar("T")


async def async_batch(
    async_iterable: AsyncIterator["T"], n: int
) -> AsyncIterator[List[T]]:
    iterator = aiter(async_iterable)
    while True:
        try:
            yield (anext(iterator) for i in range(n))
        except StopAsyncIteration:
            break
```

The library [aiostream](https://aiostream.readthedocs.io/en/stable/) provides similar tools for lower versions of Python, but using it would introduce an additional dependency.

Reply:
Good thinking about `aiter` and `anext`, but the matter of the last batch would make the code end up pretty similar to the current implementation then (since you'd need to keep a list of all the items of the current batch in any case).

Reply:
Thanks for the clarification.